### ISSUE TITLE
iOS app crashing on device at startup with .NET 8 preview 5

### ISSUE THREAD

**Issue by jmezach** [dotnet/maui#15673]
### Description

It seems that with .NET 8 preview 5 (and possibly earlier previews) a brand-new .NET MAUI iOS app crashes at startup on an iOS Device running iOS 12.5.5.

### Steps to Reproduce

1. Run dotnet new maui -o MauiApp1
2. Run dotnet build -t:Run -f net8.0-ios -p:RuntimeIdentifier=ios-arm64 -p:_DeviceName=<device-id-of-ios-device-with-12.5.5>
3. App crashes with the following error: error: * Assertion at /Users/runner/work/1/s/src/mono/mono/utils/mono-hwcap-arm64.c:28, condition `res == 0' not met

### Link to public reproduction project repository

https://github.com/jmezach/MauiApp8CrashRepro

### Version with bug

8.0.0-preview.5.8529

### Last version that worked well

7.0.86

### Affected platforms

iOS

### Affected platform versions

iPhone OS 12.5.5

**Comment by drasticactions** [comment]
Did you try deploying a non-MAUI UI iOS app with that SDK to see what it does? 

Try running

`dotnet workload install ios`
`dotnet new ios`

and run the resulting program.

**Comment by drasticactions** [comment]
CC @rolfbjarne, have you seen this before?

**Comment by jmezach** [comment]
@drasticactions I'll have to get back to you on that next week as I'm currently working from home and don't have the iOS device running this version with me, it is in the office.

@rolfbjarne That issue specifically talks about Mac Catalyst which is a different OS afaik than iOS.

**Comment by rolfbjarne** [comment]
> @rolfbjarne That issue specifically talks about Mac Catalyst which is a different OS afaik than iOS.

That's correct, but they're very similar, and that code is specific to ARM64, which applies to both iOS and Mac Catalyst.

It's also exactly the assert message I was getting on Mac Catalyst.

**Comment by jmezach** [comment]
@rolfbjarne Thought so, just wanted to check. We'll give it another try when preview 6 ships to see if that fixes our issue and report back here. Thanks for the quick reply!

**Comment by gabsamples6** [comment]
@rolfbjarne @drasticactions Hi not sure if same issue but any new iOS app I run in preview 5 .net 8 attached to a physical device I get error. Just launch a brand new app from template in iOS attached to a physical device and I get
![image](https://github.com/dotnet/maui/assets/109357537/9def66fe-4791-4ed9-a0e6-473769387b77)
 
Is it me ? Can you run any iOS attached to a device?

**Comment by rolfbjarne** [comment]
> Hi not sure if same issue but any new iOS app I run in preview 5 .net 8 attached to a physical device I get error. Just launch a brand new app from template in iOS attached to a physical device and I get

That's a known issue in preview 5 that we unfortunately were not able to fix before it went out. It'll be fixed in preview 6.

**Comment by gabsamples6** [comment]
@rolfbjarne Thank you for your prompt reply - these days is very much appreciated

**Comment by jmezach** [comment]
We just tried preview 6 and can confirm that this issue appears to be fixed. However we've found a new bug as described in #16097.
